src/csharp.ts

```diff
@@ -0,0 +1,90 @@
+export namespace CSharp {
+	export function parseClasses(documentText: string): ParsedClass[] {
+		let matches = documentText.matchAll(/public class (\w+)[^{]*{((?:[^}{]+|{(?:[^}{]+|{[^}{]*})*})*)}/g);
```

Comment:
Trying to understand this regex. It looks like it checks for possibly 3 nested braces? I get the first for the class scope and the 2nd for the `{get; set;}` function, but I don't understand the need for the 3rd except if a class was being defined inside a class?

Reply:
This is a placeholder until I get something better. If a class has some function definitions with more nested scopes, it could currently break it. I need to make a matching brace one.

Reply:
For the regex statements, should we replace spaces with `/s+` Since 
```
public      class       


TestClass



{
```

Is ugly, but valid syntax.

Reply:
At this point I'm starting to wonder if the best route would be tokenizing the class...
Because then we could have the confidence to correctly parse and hopefully cover any edge cases we might miss with the regex.
```csharp
public class /* I really like this class */ TestClass {

}
```